app/controllers/listings_controller.rb

```diff
@@ -581,4 +594,31 @@ def get_transaction_process(community: community, listing: listing)
         raise ArgumentError.new("Can not find transaction process: #{opts}") if process.nil?
       }
   end
+
+  def shapes_api
+    ListingService::API::Api.shapes
+  end
```

Comment:
For global searching in dynamic language I noticed this pattern makes it difficult to find where an api method is used in the codebase. But if we instead do:
```ruby
def listings_api
  ListingsService::API::Api
end

listings_api.shapes.get(...)
```
then you can still do global search for shapes.get( to find usages.